_posts/2017-10-06-styling-react-app.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: Styling a React application
+date: 2017-10-06
+---
+
+In the last couple of weeks I looked at [how to write a simple React application from scratch](https://caroleolivier.github.io/blog/2017/09/22/creating-a-react-app-from-scratch) and [how to test it](https://caroleolivier.github.io/blog/2017/09/26/testing-a-react-app). 
+Today I looked at how to style it.
+
+I have limited experience of styling websites (I have only used CSS and SASS in the past) so I googled the subject and found many, many solutions. I won't walk you through all of them as this is not the subject of this post, I will only describe the solution I chose and how I set it up. If you would like to know more I highly recommend this [post](http://andrewhfarmer.com/how-to-style-react/) by Andrew Farmer, it is very informative and very well written.
+<br/>
+So after reading up a bit about the different solutions I settled for [CSS Modules](https://github.com/css-modules/css-modules) together with [SASS](http://sass-lang.com/) and here is how I set it up in my web application.
+
+
+#### CSS Modules
+
+I am new to [CSS Modules](https://github.com/css-modules/css-modules) but I really like the idea behind it:
+> A CSS Module is a CSS file in which all class names and animation names are scoped locally by default.
+
+CSS classes are scoped globally by default so when your application grows it quickly becomes a nightmare to maintain and keep track of which classes is used where. With CSS modules (and the help of your building engine) you can import CSS files within a JavaScript file and CSS classes will automatically be scoped locally to that file (if I am not clear you will see in the setup section what I mean).
+<br/>
+In the context of React this means that you can easily scope a CSS file to a React component. This goes really well with React philosophy where "componentisation" and isolation is a big deal. By default with React we already define together both the structure (JSX) and logic (JavaScript) of a component so by using CSS Modules we are adding the style to the mix and completes the definition of the component and I think that's really cool!
```

Comment:
By default with React we already define ~together~ both the structure (JSX) and logic (JavaScript) of a component together so_,_ by using CSS Modules_,_ we are adding the style to the mix _which_ completes the definition of the component; and I think that's really cool